### PR TITLE
refactor: spotlight

### PR DESCRIPTION
**Pull request by mafewtm**
## Description

This refactors the spotlight mechanic to use ox_lib caching and keybinds with statebags

## Checklist

- [x] I have personally loaded this code into an updated Qbox project and checked all of its functionality.
- [x] My pull request fits the contribution guidelines & code conventions.

### CODE REVIEW

File: client/main.lua [hunk 1]
```diff
@@ -288,11 +272,40 @@ local function handleInVehicle()
     end
 end
 
-AddEventHandler('ox_lib:cache:vehicle', function()
+local spotlight = lib.addKeybind({
+    name = 'helicam',
+    description = 'Toggle Helicopter Spotlight',
+    defaultKey = '7',
+    dsiabled = true,
```
Comment (Manason) [inline client/main.lua]:
```suggestion
    disabled = true,
```

File: client/main.lua [hunk 2]
```diff
@@ -288,11 +272,40 @@ local function handleInVehicle()
...
+})
+
+---@diagnostic disable-next-line: param-type-mismatch
+AddStateBagChangeHandler('spotlight', nil, function(bagName, key, value)
+    local entity = GetEntityFromStateBagName(bagName)
+
+    SetVehicleSearchlight(entity, value, false)
+end)
+
+lib.onCache('vehicle', function(vehicle)
```
Comment (Manason) [inline client/main.lua]: This should be onCache('seat') instead to capture the edge case of someone switching seats from rear passenger to pilot.

File: client/main.lua [hunk 3]
```diff
@@ -288,11 +272,40 @@ local function handleInVehicle()
...
+        TriggerServerEvent('qbx_helicam:server:toggleSpotlightState', netId)
+    end,
+})
+
+---@diagnostic disable-next-line: param-type-mismatch
+AddStateBagChangeHandler('spotlight', nil, function(bagName, key, value)
+    local entity = GetEntityFromStateBagName(bagName)
+
+    SetVehicleSearchlight(entity, value, false)
+end)
```
Comment (Manason) [inline client/main.lua]: No gate conditions for the value?

**Review by Manason — CHANGES_REQUESTED** [review]

File: client/main.lua (hunk 3 above)
Comment (mafewtm) [reply client/main.lua]: Why do we need a gate condition? The value is either true or false and it sets the native based on that value. If you were to put in a different value other than a boolean, it wouldn't work anyway.

File: client/main.lua [hunk 4]
```diff
@@ -287,19 +287,21 @@ local spotlight = lib.addKeybind({
...
+AddStateBagChangeHandler('spotlight', nil, function(bagName, _, value)
     local entity = GetEntityFromStateBagName(bagName)
 
     SetVehicleSearchlight(entity, value, false)
 end)
 
-lib.onCache('vehicle', function(vehicle)
-    local model = GetEntityModel(vehicle)
+lib.onCache('seat', function(seat)
+    if not cache.vehicle then return end
```
Comment (Manason) [inline client/main.lua]: in this case, need to disable the spotlight keybind again, right?

**Review by Manason — APPROVED** [review]

**Pull request merged**